pkg/sources/sources.go

```diff
@@ -104,39 +104,81 @@ func NewConfig(opts ...func(*Config)) Config {
 	return *c
 }
 
+// Counter is a simple monotonically-increasing counter for tracking progress.
+// It includes a successCnt that records the number of successful scans.
+// It also includes a totalCnt that records the total number of scans.
+// This is necessary in order to display the number of concurrent scans
+// regardless of whether they are successful or not.
+// successCnt is the only field used when calculating the percentage complete.
```

Comment:
**Question:** Same question about calculating percentage.. I think we should increase the percent even if it failed to indicate we are progressing.

Reply:
Would you mind clarifying this statement? I'm not sure I follow.

Reply:
Sorry for poor wording.. I edited it with some commas. But I was referring to this:

> successCnt is the only field used when calculating the percentage complete

If we are scanning 10 repos and one of them fails, I think the percent complete should be 100%, not 90%.

Reply:
Oh interesting. I was under the impression percent complete can only be 100% if there are no failures.

Reply:
Because i guess that way in the UI it doesn't look like a success?

Reply:
Maybe i confused healthy w/ complete

Reply:
Hm, yeah. That's the thread that spawned the idea of "scan reports" which will more accurately capture which items were successfully scanned.

In the mean time, I'm not sure which we should go with.. I still think from a user perspective I would want to see the progress increase even if a repo failed to scan, but I agree that the entire scan shouldn't be considered healthy in that case.

Reply:
Hmm so I think maybe we need to wait until we have scan reports because I'm not so sure this PR helps. I think we still end at the same place of 99/100 repos fail to scan we still would get 100% scan completed progress which is probably less helpful to the user.